packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -499,6 +504,16 @@ export const useField = <
     setSelectedSections: (activeSectionIndex) => setSelectedSections(activeSectionIndex),
   }));
 
+  const handleClearValue = useEventCallback((event: React.MouseEvent, ...args) => {
+    // the click event on the endAdornmnet would propagate to the input and trigger the `handleInputClick` handler.
+    event.stopPropagation();
```

Comment:
I no longer remember if this was necessary for something, but ideally, we need to avoid calling `stopPropagation`. 💡 
Source: https://github.com/mui/material-ui/pull/38072#issuecomment-1677775166

Reply:
In this case, the click on the button propagates to the input, and triggers `handleInputClick` and messes with the focus. If we want the focused section after clear to be the first one, this seemed like a decent solution, although I agree it can be messy.

https://github.com/mui/mui-x/assets/72460825/ef2c9d6b-15bf-4732-b609-d0531e778dd1

Reply:
Thanks! This seems like a good solution 🎉

Reply:
Great! 👏 
I'm glad that it seems to work out. 👍